netrd/distance/hamming.py

```diff
@@ -55,9 +55,34 @@ def dist(self, G1, G2):
         .. [1] https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.hamming.html#scipy.spatial.distance.hamming
 
         """
+
+        if G1.number_of_nodes() == G2.number_of_nodes():
+            N = G1.number_of_nodes()
+        else:
+            raise ValueError("Graphs have the same number of nodes")
+
         adj1 = nx.to_numpy_array(G1)
         adj2 = nx.to_numpy_array(G2)
-        dist = scipy.spatial.distance.hamming(adj1.flatten(), adj2.flatten())
-        self.results['dist'] = dist
-        self.results['adjacency_matrices'] = adj1, adj2
+
+        # undirected case: consider only upper triangular
+        mask = np.triu_indices(N, k=1)
+
+        # directed case: consider all but the diagonal
+        if nx.is_directed(G1) or nx.is_directed(G2):
+            new_mask = np.tril_indices(N, k=-1)
+            mask = (np.append(mask[0], new_mask[0]), np.append(mask[1], new_mask[1]))
+
+        # only if there are self-loops include the diagonal
+        # this corrects the implicit denominator of Hamming, which
+        # should be N^2 for networks with self-loops and N(N-1) for
+        # those without
+        if next(nx.selfloop_edges(G1), None) or next(nx.selfloop_edges(G2), None):
```

Comment:
Why `None` instead of `False`? Seems more appropriate inside a conditional

Reply:
Agree, and where I test for selfloops elsewhere in the code I use `None`. I'll tweak it for consistency.